code/modules/mob/living/simple_animal/slime/slime.dm

```diff
@@ -510,3 +521,36 @@
 
 /mob/living/simple_animal/slime/random/Initialize(mapload, new_colour, new_is_adult)
 	. = ..(mapload, pick(slime_colours), prob(50))
+
+/mob/living/simple_animal/slime/apply_damage(damage = 0,damagetype = BRUTE, def_zone = null, blocked = FALSE, forced = FALSE)
+	if(damage && damagetype == BRUTE && !forced && (transformeffects & SLIME_EFFECT_ADAMANTINE))
+		blocked += 50
+	. = ..(damage, damagetype, def_zone, blocked, forced)
+
+/mob/living/simple_animal/slime/attack_ghost(mob/user)
+	if(transformeffects & SLIME_EFFECT_LIGHT_PINK)
+		humanize_slime(user)
+
+/mob/living/simple_animal/slime/proc/humanize_slime(mob/user)
```

Comment:
`humanize_slime` is kind of a bad name for it, it implies you are turning the slime into a human not having a ghost assume control of the slime.